### PR TITLE
Cds 925 search improvements

### PR DESCRIPTION
**Pull request by K-Bains**

### CODE REVIEW

File: NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs [hunk 1]
```diff
@@ -13,12 +13,44 @@ public static DocumentClient CreateDocumentClient()
...
+            }
+            catch (Exception e)
+            {
+                throw;
+            }
+
+            if(string.IsNullOrWhiteSpace(connectionString))
+                throw new ArgumentNullException();
+
+            var endPoint = connectionString.Split(new[] { "AccountEndpoint=" }, StringSplitOptions.None)[1]
```
Comment (NickGraham101) [inline NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs]: Will this throw a useful error if the connection string is badly formatted and the split doesn't work as expected?  If not it should be wrapped in a try catch

File: NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs [hunk 2]
```diff
@@ -13,12 +13,44 @@ public static DocumentClient CreateDocumentClient()
...
+                .Split(';')[0]
+                .Trim();
+
+            if (string.IsNullOrWhiteSpace(endPoint))
+                throw new ArgumentNullException();
+
+            var key = connectionString.Split(new[] { "AccountKey=" }, StringSplitOptions.None)[1]
```
Comment (NickGraham101) [inline NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs]: Will this throw a useful error if the connection string is badly formatted and the split doesn't work as expected? If not it should be wrapped in a try catch

File: NCS.DSS.Customer/SearchCustomerHttpTrigger/Function/SearchCustomerHttpTrigger.cs [hunk 3]
```diff
@@ -53,22 +59,52 @@ public static class SearchCustomerHttpTrigger
                 return HttpResponseMessageHelper.NoContent();
             }
 
+            if (!string.IsNullOrEmpty(givenName))
+            {
+                query += "GivenName:(" + givenName + "* OR " + givenName + ") ";
```
Comment (NickGraham101) [inline NCS.DSS.Customer/SearchCustomerHttpTrigger/Function/SearchCustomerHttpTrigger.cs]: Isn't String.Format preferred?

File: NCS.DSS.Customer/SearchCustomerHttpTrigger/Service/SearchCustomerHttpTriggerService.cs [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
     public class SearchCustomerHttpTriggerService : ISearchCustomerHttpTriggerService
     {
-        public async Task<List<Models.Customer>> SearchCustomerAsync(string givenName = null, string familyName = null, string dateofBirth = null,
-            string uniqueLearnerNumber = null)
+        public async Task<List<Models.Customer>> SearchCustomerAsync(ISearchIndexClient indexClient, string searchText,
+            string filter = null, IList<string> order = null, IList<string> facets = null)
         {
             var documentDbProvider = new DocumentDBProvider();
-            var customer = await documentDbProvider.SearchAllCustomer(givenName, familyName, dateofBirth, uniqueLearnerNumber);
+            var customer = await documentDbProvider.SearchCustomer(indexClient, searchText, filter, order, facets);
```
Comment (NickGraham101) [inline NCS.DSS.Customer/SearchCustomerHttpTrigger/Service/SearchCustomerHttpTriggerService.cs]: Should this be wrapped in a try catch or is there one higher up?

Comment (K-Bains) [reply NCS.DSS.Customer/SearchCustomerHttpTrigger/Service/SearchCustomerHttpTriggerService.cs]: Theres a try catch on the DB Provider side where it call Azure Search

File: NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs (hunk 2 above)
Comment (K-Bains) [reply NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs]: Missed a checkin, will sort this out now

File: NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs (hunk 1 above)
Comment (K-Bains) [reply NCS.DSS.Customer/Cosmos/Client/DocumentDBClient.cs]: missed commit

**Pull request closed without merging**